### PR TITLE
[fix] Better sklearn imports

### PR DESCRIPTION
**Pull request by paxcema**
This PR is related to [Lightwood#1206](https://github.com/mindsdb/lightwood/pull/1206). Changes are:

- Moved `sklearn.metrics` imports from being dynamic within a module to static in the module initialization
- Added a label encoder so that the evaluator can handle string labels, provided a list of expected categories is passed when calling.

### CODE REVIEW

File: mindsdb_evaluator/accuracy/__init__.py [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# flake8: noqa  # otherwise F401 is triggered when adding all metrics from sklearn, even though no overwrites happen
+import importlib
+from sklearn.metrics import __all__ as __sklearn_accs__
```
Comment (pedrofluxa) [inline mindsdb_evaluator/accuracy/__init__.py]: This seems slightly hacky to me.

File: mindsdb_evaluator/accuracy/__init__.py [hunk 2]
```diff
@@ -7,6 +11,13 @@
     evaluate_cat_array_accuracy, \
     complementary_smape_array_accuracy
 
+
+__sklearn_accs__ = [acc for acc in __sklearn_accs__ if acc[0].lower() == acc[0]]
+
+sk_metrics = importlib.import_module('sklearn.metrics')
```
Comment (pedrofluxa) [inline mindsdb_evaluator/accuracy/__init__.py]: This too. Not sure if it is a good idea to add modules from external sources to `globals()` I know it is sklearn, but in the future other developers might look a this code and try the same with other libraries of unknown origin. 

Just my opinion here, the code should work just fine.

**Review by pedrofluxa — APPROVED** [review]
Seems OK, though the way the metrics get integrated into the evaluator seems a little hacky to me. I understand it allows for a smoother integration with sklearn but maybe, just maybe, a good old long and boring import x,y,z would do the trick.

**Comment by paxcema** [comment]
It's a good point @pedrofluxa, I was going for coverage but agreed it looks "hacky". I can tone it down and import a good initial set. If users want more, we can manually add them 👍
